### ISSUE TITLE
Azure static web apps support certificate instead of secret for custom AAD authentication.

### ISSUE THREAD

**Issue by duggaraju** [Azure/static-web-apps#1443]
**Is your feature request related to a problem? Please describe.**
Static web apps can be configured with custom authentication. But that requires creation of a secret on the Entra ID application.  as described [here](https://learn.microsoft.com/en-us/azure/static-web-apps/authentication-custom?tabs=aad%2Cinvitations)

**Describe the solution you'd like**
Similar to key vault if we can use a certificate and the certificate can be directly requested form the key vault (similar to secret) then rotation becomes simpler.

**Describe alternatives you've considered**
No alternatives.

**Additional context**
For example we cannot use Azure Static web Apps as Microsoft 1st Party app because 1st party apps only support certificates but not secrets.

**Comment by molo76** [comment]
I second @duggaraju's request above, this would remove the need for secrets on the Entra ID app registration. Without this feature we cannot satisfy security policy requirements to eliminate Entra ID app secrets.

**Comment by fabge** [comment]
second this. even better would be federated credentials

**Comment by davidzenisu** [comment]
Yes, agreed. Seeing as this was [just released](https://learn.microsoft.com/en-us/entra/workload-id/workload-identity-federation-config-app-trust-managed-identity?tabs=microsoft-entra-admin-center) for managed identities in general, it should be possible for SWA without major implementation efforts (I would hope)

**Comment by fabge** [comment]
not sure about that, [the documentation](https://learn.microsoft.com/en-us/entra/workload-id/workload-identity-federation-config-app-trust-managed-identity?tabs=microsoft-entra-admin-center#microsoftidentityweb) shows the following code for Microsoft.Identitiy.Web:

```json
{
  "AzureAd": {
    "Instance": "https://login.microsoftonline.com/",
    "ClientId": "YOUR_APPLICATION_ID",
    "TenantId": "YOUR_TENANT_ID",

   "ClientCredentials": [
      {
        "SourceType": "SignedAssertionFromManagedIdentity",
        "ManagedIdentityClientId": "YOUR_USER_ASSIGNED_MANAGED_IDENTITY_CLIENT_ID",
        "TokenExchangeUrl":"api://AzureADTokenExchange"
      }
   ]
  }
}
```

Anything like this is currently unsupported by the [staticwebapp.config.json](https://learn.microsoft.com/en-us/azure/static-web-apps/authentication-custom?tabs=aad%2Cinvitations#configure-a-custom-identity-provider) of Azure Static Web Apps...
